nucleus/admin/server-mgmt/src/main/java/fish/payara/admin/servermgmt/cli/PrintCertificateCommand.java

```diff
@@ -0,0 +1,255 @@
+ *
+ *     Modifications:
+ *     If applicable, add the following below the License Header, with the fields
+ *     enclosed by brackets [] replaced by your own identifying information:
+ *     "Portions Copyright [year] [name of copyright owner]"
+ *
+ *     Contributor(s):
+ *     If you wish your version of this file to be governed by only the CDDL or
+ *     only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *     elects to include this software in this distribution under the [CDDL or GPL
+ *     Version 2] license."  If you don't indicate a single choice of license, a
+ *     recipient has the option to distribute your version of this file under
+ *     either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *     its licensees as provided above.  However, if you add GPL Version 2 code
+ *     and therefore, elected the GPL Version 2 license, then the option applies
+ *     only if the new code is made subject to such option by the copyright
+ *     holder.
+ */
+package fish.payara.admin.servermgmt.cli;
+
+import com.sun.enterprise.admin.cli.CLICommand;
+import com.sun.enterprise.admin.servermgmt.KeystoreManager;
+import com.sun.enterprise.security.auth.realm.certificate.CertificateRealm;
+import com.sun.enterprise.util.StringUtils;
+
+import java.io.File;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.Provider;
+import java.security.PublicKey;
+import java.security.Security;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.DSAKey;
+import java.security.interfaces.RSAKey;
+import java.util.Collection;
+
+import javax.security.auth.x500.X500Principal;
+
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.CommandException;
+import org.glassfish.api.admin.CommandValidationException;
+import org.glassfish.hk2.api.PerLookup;
+import org.jvnet.hk2.annotations.Service;
+
+/**
+ * Prints information about a certificate given in a file.
+ * Uses RFC2253 for the principal distinguished name.
+ * <p>
+ * <b>WARNING:</b> JKS and JCEKS may be removed from some JDKs, then this command may fail.
+ *
+ * @author David Matejcek
+ */
+@Service(name = "print-certificate")
+@PerLookup
+public class PrintCertificateCommand extends CLICommand {
+
+    @Param
+    String file;
+
+    @Param(optional = true, defaultValue = "")
+    String certificateAlias;
+
+    @Param(optional = true)
+    String providerClass;
+
+    private Provider provider;
+
+    private File derFile;
+
+    private File keystoreFile;
+
+    private char[] keystorePassword;
+
+    @Override
+    protected void validate() throws CommandException, CommandValidationException {
+        super.validate();
+        if (!ok(this.file)) {
+            throw new CommandValidationException("The file with the certificate must be specified.");
+        }
+        final File sourceFile = new File(this.file);
```

Comment:
Not 100% sure about our style guide but from what I did I'd assume we do use `this` when we assign to a field but do not use it when reading a field unless the field is shadowed by a local variable.

Reply:
I think it is good to always have some visual difference between fields and local variables. It forces to think about the context of the variable on the first sight. Payara code conventions page tells nothing about this...

Reply:
Sure. Most IDEs will already make it visually very different right out of the box and if not you could configure it. Its basically the same argument people had to prefix every field with underscore or `m_`. At least I am happy we have IDEs now that don't make us prefix stuff.

Reply:
The problem in IDEs is that most of the code in the world has so many issues that one more warning is simply ignored.
I never used prefixes, because they change too many things, finally the readibility would be worse than better.
The `this` does not change anything in compiled class, it is only about syntax. It is so obvious for me, because it helped to cleanup classes with 5000 lines, hundred variables, one method ... yes, that's not an issue in Payara, so here you don't see any advantage of `this`.
I think more similar discussion is about using `final` ...